refinery/tool_manager/models.py

```diff
@@ -548,22 +548,23 @@ def _get_detailed_nodes_dict(self, node_uuid_list,
             - A full url pointing to our Node's FileStoreItem's datafile
         """
         solr_response_json = get_solr_response_json(node_uuid_list)
-        node_info = {
-            node["uuid"]: {
+        node_info = {}
+        for node in solr_response_json["nodes"]:
+            node_info[node["uuid"]] = {
                 self.NODE_SOLR_INFO: node,
                 self.FILE_URL: get_file_url_from_node_uuid(
                     node["uuid"],
                     require_valid_url=require_valid_urls
                 ),
                 self.AUXILIARY_FILE_LIST: [
                     get_file_url_from_node_uuid(
-                        child.uuid, require_valid_url=require_valid_urls
-                    ) for child in Node.objects.get(uuid=node["uuid"]).
-                    get_auxiliary_nodes()
+                        uuid, require_valid_url=require_valid_urls
+                    ) for uuid in
+                    Node.objects.get(
+                        uuid=node["uuid"]
+                    ).get_auxiliary_node_uuids()
```

Comment:
Could you put this outside the list comprehension for clarity: `auxiliary_nodes = Node.objects.get(uuid=node["uuid"]).get_auxiliary_nodes()`?